newdle/client/src/components/AnswerCalendar.js

```diff
@@ -0,0 +1,159 @@
+import _ from 'lodash';
+import {Grid} from 'semantic-ui-react';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {serializeDate, toMoment} from '../util/date';
+import {useSelector} from 'react-redux';
+import styles from './Answer.module.scss';
+import AnswerCalendarDay from './AnswerCalendarDay';
+import {getNewdleTimeslots, getNewdleDuration} from '../selectors';
+
+const OVERFLOW_HEIGHT = 0.5;
+
+function calculateHeight(start, end, minHour, maxHour) {
+  let startMins = start.hours() * 60 + start.minutes();
+  let endMins = end.hours() * 60 + end.minutes();
+
+  if (startMins < minHour * 60) {
+    startMins = minHour * 60;
+  }
+  if (endMins > maxHour * 60) {
+    endMins = maxHour * 60;
+  }
+  const height = ((endMins - startMins) / ((maxHour - minHour) * 60)) * 100;
+  return height > 0 ? height : OVERFLOW_HEIGHT;
+}
+
+function calculatePosition(start, minHour, maxHour) {
+  const spanMins = (maxHour - minHour) * 60;
+  let startMins = start.hours() * 60 + start.minutes() - minHour * 60;
+
+  if (startMins < 0) {
+    startMins = 0;
+  }
+
+  const position = (startMins / spanMins) * 100;
+  return position < 100 ? position : 100 - OVERFLOW_HEIGHT;
+}
+
+function groupOverlaps(timeSlots) {
+  const sortedTimeSlots = _.orderBy(timeSlots, ['pos'], ['asc']);
+  // marks items that overlap with its successors
+  let cluster_id = 0;
+  const clusteredCandidates = sortedTimeSlots.map((candidate, index, sortedTimeSlots) => {
+    if (
+      sortedTimeSlots[index - 1] &&
+      moment(candidate.startTime, 'HH:mm').isSameOrAfter(
+        moment(sortedTimeSlots[index - 1].endTime, 'HH:mm')
+      )
+    ) {
+      cluster_id += 1;
+    }
+    return {...candidate, cluster_id: cluster_id};
+  });
+
+  return _.chain(clusteredCandidates)
```

Comment:
this whole expression looks like it can be replaced by `Object.values(_.groupBy(clusteredCandidates, c => c.cluster_id))`